pkg/agent/endpoints/workload/handler_test.go

```diff
@@ -624,23 +743,93 @@ func TestFetchJWTSVID(t *testing.T) {
 		{
 			name: "success all",
 			identities: []cache.Identity{
+				identityFromX509SVID(x509SVID1Dup),
 				identityFromX509SVID(x509SVID1),
 				identityFromX509SVID(x509SVID2),
 			},
-			audience:       []string{"AUDIENCE"},
-			expectCode:     codes.OK,
-			expectTokenIDs: []spiffeid.ID{x509SVID1.ID, x509SVID2.ID},
+			audience:   []string{"AUDIENCE"},
+			expectCode: codes.OK,
+			expectedResp: []expectedSVID{
+				{
+					spiffeID: x509SVID1Dup.ID.String(),
+					hint:     "external",
+				},
+				{
+					spiffeID: x509SVID1.ID.String(),
+					hint:     "internal",
+				},
+				{
+					spiffeID: x509SVID2.ID.String(),
+				},
+			},
 		},
 		{
 			name: "success specific",
 			identities: []cache.Identity{
 				identityFromX509SVID(x509SVID1),
 				identityFromX509SVID(x509SVID2),
 			},
-			spiffeID:       x509SVID2.ID.String(),
-			audience:       []string{"AUDIENCE"},
-			expectCode:     codes.OK,
-			expectTokenIDs: []spiffeid.ID{x509SVID2.ID},
+			spiffeID:   x509SVID2.ID.String(),
+			audience:   []string{"AUDIENCE"},
+			expectCode: codes.OK,
+			expectedResp: []expectedSVID{
+				{
+					spiffeID: x509SVID2.ID.String(),
+				},
+			},
+		},
+		{
+			name:       "identities with duplicated hints",
+			identities: identities,
+			audience:   []string{"AUDIENCE"},
+			expectCode: codes.OK,
+			expectedResp: []expectedSVID{
```

Comment:
These results look incorrect to me. The test setup looks pretty contrived because `x509SVID1` and `x509SVID1Dup` have the same Entry ID because the Entry ID is being set to the SPIFFE ID, which is the same for both of these X.509-SVIDs. In practice all entries should have unique entry IDs, in which case `x509SVID1Dup` should be included in the results. 

Can we set explicit entry IDs for each SVID in the tests rather than reusing the entry's SPIFFE ID to make this less ambiguous?

Reply:
You're right, this test setup is unrealistic, but the x509SVID1Dup is not being returned here because we do not include it in the identities fake response. I'm going to update it to include it in the fake response, and to use unique entry IDs :+1:

Reply:
[a18dfea](https://github.com/spiffe/spire/pull/3993/commits/a18dfeae82834846d54f3d7f7f1a545fb8e2aec1)